### PR TITLE
Caleb lab

### PR DESCRIPTION
**Pull request by aesthetiques**

### CODE REVIEW

File: caleb-lab/server.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  if(req.method === 'POST'){
+    if(req.url.pathname === '/cowsay'){
+      bodyParser(req, function(err){
+        if(err) console.error(err);
+        let message = cowsay.think({text: req.body.text});
+        console.log(res.body);
+        res.writeHead(200, {'Content-Type': 'text/plain' });
+        res.write(message);
+        res.end(); // this will not be here, will be in asynchronous callback(s);
```
Comment (JudyVue) [inline caleb-lab/server.js:23]: **TODO FOR LOST POINTS**
I won't comment on every comment and console.log, but a reminder to please take them out before submission.

File: caleb-lab/server.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      });
+    }else{
+      let message = cowsay.say({text: '400: bad request\n try localhost:3000/cowsay with a proper body'});
+      res.writeHead(400, {'Content-Type': 'text/plain' });
+      res.write(message);
+      res.end();
+      // badrequest
+    }
+  }
+  if(req.method === 'GET'){
```
Comment (JudyVue) [inline caleb-lab/server.js:42]: **TODO FOR LOST POINTS**
While your GET method is good, it doesn't exactly fall in line with the directions, which are:

- the response header should include Content-Type: text/plain
- if the json {text: message} is set in the body, respond with:
- a status code of 200
- a body including the value returned from cowsay.say({text: <querystring text>})
- if the json{text: message}is not set in the body, respond with:
- status code = 400
- a body including the value returned from cowsay.say({text: 'bad request\ntry: localhost:3000/cowsay?text=howdy'})

In other words, I should be able to use the command:

```
http GET :3000/?text=hello
```
to get a cow that says hello, and I don't get that with the way your code is written.

File: caleb-lab/test/test.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
+'use strict';
```
Comment (JudyVue) [inline caleb-lab/test/test.js:1]: Good test module.

File: caleb-lab/lib/body-parser.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (JudyVue) [inline caleb-lab/lib/body-parser.js:1]: **TODO FOR LOST POINTS**

A top-level comment to let you know you're missing some of these bullet points:

- add a package.json
- add a eslintrc
- add a gitignore
- add a readme with project description
- include any npm scripts for starting server, linting, testing, etc
- test your code
- ensure that all of your methods have test coverage
- write tests which start your server, send and receive, and confirm functionality
